src/extension.ts

```diff
@@ -1,30 +1,166 @@
+import { FileDecorationProvider } from "./FileDecorationProvider";
 import { SidebarProvider } from "./SidebarProvider";
+import {
+  SOCKET_URL,
+  USER_READY,
+  FILE_CLICK,
+  CURSOR_MOVE,
+  DISCONNECT,
+  JOIN_SESSION,
+} from "./utils/constants";
+import { io } from "socket.io-client";
 import * as vscode from "vscode";
 
-// This method is called when your extension is activated
-// Your extension is activated the very first time the command is executed
+// This method is called when your extension is activated.
+// Currently this is activated as soon as user open VSCode
 export function activate(context: vscode.ExtensionContext) {
+  // Autosave
+  const config = vscode.workspace.getConfiguration("files");
+  config.update("autoSave", "afterDelay", true);
+  config.update("autoSaveDelay", 100, true);
 
+  // Initialize variables
+  const workspace = vscode.workspace.workspaceFolders?.[0];
+  const sessionId = workspace?.name || "";
```

Comment:
TODO: this should eventually hook up the API [here](https://github.com/hatchways/live-session-manager/pull/1/files#diff-b699541a212b69238da531a570d818565757eb22637899edbc8d56b99eae01c1R14)

Reply:
Hmm, this actually wouldn't be calling the API, but instead it's more likely that it would be passed in as an environment variable. So, I would change this to use an env var called `SESSION_ID` instead, and confirm it's the right one when you pass it in via docker. You might want to set up an env var file of some sort, because there will be other actions using env vars later.